hcaptcha/src/main/java/com/stripe/hcaptcha/HCaptcha.kt

```diff
@@ -0,0 +1,146 @@
+package com.stripe.hcaptcha
+
+import android.util.AndroidRuntimeException
+import androidx.annotation.RestrictTo
+import androidx.fragment.app.FragmentActivity
+import com.stripe.hcaptcha.config.HCaptchaConfig
+import com.stripe.hcaptcha.config.HCaptchaInternalConfig
+import com.stripe.hcaptcha.config.HCaptchaSize
+import com.stripe.hcaptcha.task.Task
+import com.stripe.hcaptcha.webview.HCaptchaHeadlessWebView
+
+/**
+ * hCaptcha client which allows invoking of challenge completion and listening for the result.
+ *
+ * <pre>
+ * Usage example:
+ * 1. Get a client either using the site key or customize by passing a config:
+ * client = HCaptcha.getClient(this).verifyWithHCaptcha(YOUR_API_SITE_KEY)
+ * client = HCaptcha.getClient(this).verifyWithHCaptcha([com.stripe.hcaptcha.HCaptchaConfig])
+ *
+ * // Improve cold start by setting up the hCaptcha client
+ * client = HCaptcha.getClient(this).setup([com.stripe.hcaptcha.HCaptchaConfig])
+ * // ui rendering...
+ * // user form fill up...
+ * // ready for human verification
+ * client.verifyWithHCaptcha();
+ * 2. Listen for the result and error events:
+ *
+ * client.addOnSuccessListener(new OnSuccessListener&lt;HCaptchaTokenResponse&gt;() {
+ * @Override
+ * public void onSuccess(HCaptchaTokenResponse response) {
+ * String userResponseToken = response.getTokenResult();
+ * Log.d(TAG, "hCaptcha token: " + userResponseToken);
+ * // Validate the user response token using the hCAPTCHA siteverify API
+ * }
+ * })
+ * .addOnFailureListener(new OnFailureListener() {
+ * @Override
+ * public void onFailure(HCaptchaException e) {
+ * Log.d(TAG, "hCaptcha failed: " + e.getMessage() + "(" + e.getStatusCode() + ")");
+ * }
+ * });
+ * </pre>
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+interface IHCaptcha {
+
+    /**
+     * Constructs a new client which allows to display a challenge dialog
+     *
+     * @param config Config to customize: size, theme, locale, endpoint, rqdata, etc.
+     * @return new [HCaptcha] object
+     */
+    fun setup(config: HCaptchaConfig): HCaptcha?
+
+    /**
+     * Shows a captcha challenge dialog to be completed by the user
```

Comment:
Do we need this? Can we remove the code?

Reply:
We need the `verifyWithHCaptcha` since that's what launches the hCaptcha challenge (passive or active) that we use to collect data.

Are you suggesting we combine it with the `setup` method?

Reply:
The docs say "Shows", which leads me to believe we have more code than we need.

I think I'm suggesting 2 things: 
* Update the docs to reflect what this actually does.
* Remove any code that's not related to the passive captcha

Reply:
We're considering using active hCaptcha on some endpoints as well in the future, so I think it makes sense to leave this functionality available.

Additionally, the only difference between the active and passive captchas are the values that are passed to the webview, and how it draws its size. Eliminating that code drops a lot of functionality with very little change in complexity.

Reply:
I'll update the docstring to make this more clear.